### DISCUSSION TITLE
Switching and Validating Schemas based on attribute value

### DISCUSSION THREAD

**Discussion by mathan-mohan91** [json-schema-org/community#635] [Q&A] [answered]
Hey Folks, I have a question on creating Json Meta Schema.

High Level, this is how it looks like below

User-Schema
```json
{
"properties" : {
"type" : {"type" : "string"},
"enum" : {"type" : "array"},
"rating": {"type": "integer"}
}
```
Customer-Schema

```json
{
"properties" : {
"type" : {"type" : "string"},
"enum" : {"type" : "array"},
"comments": {"type": "string"}
}
```
I have a top level schema

```
{
"properties" : {
  "group" : {"type" : "string"},
  "properties" : {"type" : "object"} // basically it will be a Map<String, Either of user or customer schema>
}
}
```
I tried building a if else for validation

```
"if": {
"group" : {
"const" : "user" 
} 
},
"then":
"properties": {
"additionalProperties": {
"$ref" : "/user-schema"
}
},
"else" ... customer-schema
}
``` 
The above did not work as the additional Properties were not taken to consideration and still referencing the top-level schema for user and customer. 
But wanted to understand we can have 2 different schemas under the same property and validate accordingly. 

There was a similar discussion but it kinda deviated from the topic so looking for some inputs (https://github.com/orgs/json-schema-org/discussions/202)

**Comment by gregsdennis** [comment, marked as answer]
The likely issue is that your subschemas are missing a `properties`.  For example, the `if` subschemas needs a `properties` outside of `group`.  Currently it looks like `group` is a keyword, and since it's unknown it'll just be ignored, making the `if` always validate.

Also, you'll want to make `group` required in the `if` subschema if it's possible that `group` could be missing.

```
"if": {
  "properties": {
    "group" : {
      "const" : "user"
    }
  },
  "required": ["group"]
},
...
```
